internal/policy/source/source.go

```diff
@@ -65,12 +67,30 @@ type PolicyUrl struct {
 	Kind policyKind
 }
 
+// downloadCache is a concurrent map used to cache downloaded files.
+var downloadCache sync.Map
+
 // GetPolicies clones the repository for a given PolicyUrl
 func (p *PolicyUrl) GetPolicy(ctx context.Context, workDir string, showMsg bool) (string, error) {
 	sourceUrl := p.PolicyUrl()
 
 	dest := uniqueDestination(workDir, p.Subdir(), sourceUrl)
 
+	// Check if the policy has already been downloaded.
+	// If it has try to create a symlink. If that passes return the symlink path.
+	// If the symlink creation fails, fall back to downloading the policy again.
+	if d, loaded := downloadCache.LoadOrStore(sourceUrl, dest); loaded {
```

Comment:
Having a `sourceUrl` key in `downloadCache` does not guarantee that the source was indeed downloaded, the download could have failed or could be still in progress.
I think the value in `downloadCache` should be `sync.OnceValues`.

Perhaps something like:

```suggestion
	dl, _ := downloadCache.LoadOrStore(sourceUrl, sync.OnceValues(func() (string, error) {
		dest := uniqueDestination(workDir, p.Subdir(), sourceUrl)
		log.Debugf("Download cache miss: %s", sourceUrl)
	
		// Checkout policy repo into work directory.
		log.Debugf("Downloading policy files from source url %s to destination %s", sourceUrl, dest)
	
		x := ctx.Value(DownloaderFuncKey)
		if dl, ok := x.(downloaderFunc); ok {
			return dest, dl.Download(ctx, dest, sourceUrl, showMsg)
		}
		return dest, downloader.Download(ctx, dest, sourceUrl, showMsg)
	}))
	
	if dest, err := dl.(func() (string, error))(); err != nil {
		return "", err
	}
	
	if filepath.Dir(dest) != workDir {
		fs := utils.FS(ctx)
		workdirDest := uniqueDestination(workDir, p.Subdir(), sourceUrl)
		if err := fs.SymlinkIfPossible(dest, workdirDest); err != nil {
			return "", err
		}
		
		return workdirDest, nil
	}
	
	return dest, nil
```